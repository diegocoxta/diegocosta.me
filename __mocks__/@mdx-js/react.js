module.exports = {
  MDXProvider: jest.fn(({ children }) => children),
};
