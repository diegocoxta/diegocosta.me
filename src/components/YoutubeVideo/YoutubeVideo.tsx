import React from 'react';
import styled from 'styled-components';

const Content = styled.div`
  position: relative;
  padding-bottom: 56.25%;
  padding-top: 30px;
  height: 0;
  overflow: hidden;
  margin-top: 20px;

  iframe,
  object,
  embed {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
  }
`;

interface YoutubeVideo {
  uuid: string;
}

export default function YoutubeVideo(props: YoutubeVideo): React.ReactElement {
  return (
    <Content>
      <iframe
        width="560"
        height="315"
        src={`https://www.youtube.com/embed/${props.uuid}`}
        title="YouTube video player"
        frameBorder="0"
        allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
        allowFullScreen
      ></iframe>
    </Content>
  );
}
