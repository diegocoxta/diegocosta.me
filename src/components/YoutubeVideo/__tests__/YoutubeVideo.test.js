import React from 'react';
import { render } from '@testing-library/react';

import YoutubeVideo from '../YoutubeVideo';

describe('<YoutubeVideo />', () => {
  it('renders properly', () => {
    const { baseElement, getByTestId } = render(<YoutubeVideo uuid="John Doe" />);
    expect(getByTestId('youtube-player')).toHaveAttribute('src', 'https://www.youtube.com/embed/John Doe');
    expect(baseElement).toMatchSnapshot();
  });
});
