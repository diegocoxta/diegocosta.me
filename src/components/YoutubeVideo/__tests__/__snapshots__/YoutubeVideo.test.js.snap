// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<YoutubeVideo /> renders properly 1`] = `
.c0 {
  position: relative;
  padding-bottom: 56.25%;
  padding-top: 30px;
  height: 0;
  overflow: hidden;
  margin-top: 20px;
}

.c0 iframe,
.c0 object,
.c0 embed {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

<body>
  <div>
    <div
      class="c0"
    >
      <iframe
        allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
        allowfullscreen=""
        data-testid="youtube-player"
        frameborder="0"
        height="315"
        src="https://www.youtube.com/embed/John Doe"
        title="YouTube video player"
        width="560"
      />
    </div>
  </div>
</body>
`;
